cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_brute_force_cls_dpc.cpp

```diff
@@ -66,11 +67,30 @@ static infer_result<task::classification> call_daal_kernel(const context_gpu& ct
         throw internal_error{ dal::detail::error_messages::distance_is_not_supported_for_gpu() };
     }
 
-    interop::status_to_exception(daal_knn_brute_force_kernel_t<Float>().compute(
-        daal_data.get(),
-        dal::detail::get_impl(m).get_interop()->get_daal_model().get(),
-        daal_labels.get(),
-        &daal_parameter));
+    if (!dynamic_cast<brute_force_model_impl<task::classification>*>(&dal::detail::get_impl(m))) {
+        throw internal_error{ dal::detail::error_messages::unsupported_knn_model() };
+    }
+
+    const auto deserialized_model =
```

Comment:
same comments - you don't need two times make dynamic_cast

Reply:
applied